### PR TITLE
Active gravity generator can no longer be unanchored

### PR DESCRIPTION
**Pull request by TheShuEd**
## About the PR
title

## Why / Balance
image

**Changelog**
:cl:
- fix: Gravity generator can no longer be unscrewed

### CODE REVIEW

**Comment by lzk228** [comment]
should be available to unanchor when not active
image

**Comment by TheShuEd** [comment]
Now only an unpowered gravity generator can be unscrewed. This is a pretty quick patch, you need to do handling anchoring anyway

**Comment by TheShuEd** [comment]
image

**Comment by 0x6273** [comment]
> fix: Gravity generator can no longer be unscrewed

Update this to say active, and also wrench not screw

File: Content.Server/Gravity/GravityGeneratorSystem.cs [hunk 1]
```diff
@@ -28,6 +31,16 @@ public override void Initialize()
...
                 OnSwitchGenerator);
+            SubscribeLocalEvent<GravityGeneratorComponent, UnanchorAttemptEvent>(OnUnanchorAttempt);
+        }
+
+        private void OnUnanchorAttempt(Entity<GravityGeneratorComponent> ent, ref UnanchorAttemptEvent args)
+        {
+            if (TryComp<ApcPowerReceiverComponent>(ent, out var apc) && apc.Powered)
+            {
+                _popup.PopupEntity(Loc.GetString("gravity-generator-interact-failed"), ent);
+                args.Cancel();
```
Comment (0x6273) [inline Content.Server/Gravity/GravityGeneratorSystem.cs:42]: Won't this mispredict? Pretty sure wrenching is in shared.

File: Content.Server/Gravity/GravityGeneratorSystem.cs [hunk 2]
```diff
@@ -28,6 +31,16 @@ public override void Initialize()
             SubscribeLocalEvent<GravityGeneratorComponent, InteractHandEvent>(OnInteractHand);
             SubscribeLocalEvent<GravityGeneratorComponent, SharedGravityGeneratorComponent.SwitchGeneratorMessage>(
                 OnSwitchGenerator);
+            SubscribeLocalEvent<GravityGeneratorComponent, UnanchorAttemptEvent>(OnUnanchorAttempt);
+        }
+
+        private void OnUnanchorAttempt(Entity<GravityGeneratorComponent> ent, ref UnanchorAttemptEvent args)
+        {
+            if (TryComp<ApcPowerReceiverComponent>(ent, out var apc) && apc.Powered)
```
Comment (Partmedia) [inline Content.Server/Gravity/GravityGeneratorSystem.cs:39]: Why this instead of `ent.Comp.GravityActive`?

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Okay explaining my thing now that I'm awake:

Bug is gravgens can still work even when unanchored.
Step 1 is fixing it so that doesn't work anymore.
Step 2 is preventing unanchoring if that's the route we want to go down.

Only fixing step 2 means step 1 can still happen again easily (due to design or other code like singulo unanchoring or whatever unanchoring)

**Pull request closed without merging**